utils.groovy

```diff
@@ -820,4 +820,79 @@ def matrixSend(message) {
     }
 }
 
+def build_remote_image(arches, commit, url, repo, tag, registry_secret_path, secret=None, from=None,
+                        extra_args=None) {
+    def build_args = ["--git-ref", commit, "--git-url", url, "--repo", repo,
+                      "--push-to-registry", "--auth=${registry_secret_path}"
+                     ]
+    if (secret) {
+         build_args += ["--secret", secret]
+    }
+    if (from) {
+        build_args += ["--from", from]
+    }
+    parallel arches.collectEntries { arch ->
+        [arch, {
+            pipeutils.withPodmanRemoteArchBuilder(arch: arch) {
+                def final_args = build_args + ["--arch", arch, "--tag", "${tag}-${arch}"] + extra_args
+                shwrap("cosa remote-build-container ${final_args.join(' ')}")
+            }
+        }]
+    }
+}
+
+def push_manifest(arches, repo, image_tag, manifest_tag, registry_secret_path) {
+    def images = ""
+    for (arch in arches) {
+        images += " --image=docker://${repo}:${image_tag}-${arch}"
+    }
+    // arbitrarily selecting the s390x builder; we don't run this
+    // locally because podman wants user namespacing (yes, even just
+    // to push a manifest...)
+    pipeutils.withPodmanRemoteArchBuilder(arch: "s390x") {
+        shwrap("""
+        cosa push-container-manifest \
+            --auth=${registry_secret_path} --tag ${manifest_tag} \
+            --repo ${repo} ${images}
+        """)
+    }
+}
+
+def copy_image(src_image, dest_image, registry_secret_path) {
```

Comment:
Let's make `registry_secret_path` optional? The caller might be already be passing it in via `REGISTRY_AUTH_FILE` (which... we could just use that too in `build-node-image`).